src/App.tsx

```diff
@@ -1,93 +1,267 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useContext, useEffect, useMemo, useState,
+} from 'react';
+import classNames from 'classnames';
+import { AuthContext } from './components/Auth/AuthContext';
+import { Erorr } from './components/Error/Error';
+import { FilterTodoList } from './components/FilterTodoList/FilterTodoList';
+import { TodoList } from './components/TodoList/TodoList';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TypeError } from './types/ErrorType';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  addTodos,
+  deleteTodos,
+  toggleTodo,
+  renameTodo,
+} from './api/todos';
+import { FilterType } from './types/FilterType';
 
 export const App: React.FC = () => {
+  const user = useContext(AuthContext);
+
+  const [hasError, setHasError] = useState(false);
+  const [errorType, setErrorType] = useState<TypeError>(TypeError.NONE);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletedTodosIds, setDeletedTodoIds] = useState<number[]>([]);
+  const [selectedTodoIds, setSelectedTodoIds] = useState<number[]>([]);
+
+  const loadTodosFromServer = useCallback(async () => {
+    if (user) {
+      try {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      } catch {
+        setHasError(true);
+        setErrorType(TypeError.LOAD);
+      }
+    }
+  }, []);
```

Comment:
Are you sure that you have a correct dependency in the array? I think you should add the user here 
```suggestion
  const loadTodosFromServer = useCallback(async () => {
    if (user) {
      try {
        const todosFromServer = await getTodos(user.id);

        setTodos(todosFromServer);
      } catch {
        setHasError(true);
        setErrorType(TypeError.LOAD);
      }
    }
  }, [user]);
```